hasher-matcher-actioner/terraform/api/main.tf

```diff
@@ -197,6 +197,22 @@ resource "aws_apigatewayv2_route" "hma_apigateway_post" {
   target             = "integrations/${aws_apigatewayv2_integration.hma_apigateway.id}"
 }
 
+resource "aws_apigatewayv2_route" "hma_apigateway_put" {
+  api_id             = aws_apigatewayv2_api.hma_apigateway.id
+  route_key          = "PUT /{proxy+}"
+  authorization_type = "JWT"
+  authorizer_id      = aws_apigatewayv2_authorizer.hma_apigateway.id
+  target             = "integrations/${aws_apigatewayv2_integration.hma_apigateway.id}"
+}
+
+resource "aws_apigatewayv2_route" "hma_apigateway_delete" {
+  api_id             = aws_apigatewayv2_api.hma_apigateway.id
+  route_key          = "DELETE /{proxy+}"
+  authorization_type = "JWT"
+  authorizer_id      = aws_apigatewayv2_authorizer.hma_apigateway.id
+  target             = "integrations/${aws_apigatewayv2_integration.hma_apigateway.id}"
+}
+
 resource "aws_apigatewayv2_integration" "hma_apigateway" {
```

Comment:
did you figure our why we have this with `POST` but not the other verbs? If so could you document?

Reply:
Trial and error. I added an aws_apigatewayv2_integration named hma_apigateway_put and hma_apigateway_delete and hma_apigateway_get and paired them with their corresponding aws_apigatewayv2_route resource, and got an error "integration_method must be POST when integration_type is AWS_PROXY".